pkg/server/delta/v3/server.go

```diff
@@ -252,10 +267,13 @@ func (s *server) unsubscribe(resources []string, streamState *stream.StreamState
 			// * a resource is explicitly unsubscribed by name
 			// Then the control-plane must return in the response whether the resource is removed (if no longer present for this node)
 			// or still existing. In the latter case the entire resource must be returned, same as if it had been created or updated
-			// To achieve that, we mark the resource as having been returned with an empty version. While creating the response, the cache will either:
+			// To achieve that, we mark the resource as having been returned with a fake version. While creating the response, the cache will either:
 			// * detect the version change, and return the resource (as an update)
 			// * detect the resource deletion, and set it as removed in the response
-			streamState.GetResourceVersions()[resource] = ""
+			streamState.GetKnownResources()[resource] = "unsubscribed"
```

Comment:
This is for sure hacky (though not much more than before), while allowing the use of "" for the standard case of "non-existent" resource